packages/theme/src/typography.ts

```diff
@@ -1,26 +1,100 @@
-import { fontSizes } from './fontSizes';
-import { fontWeights } from './fontWeights';
-import { fontFamilies } from './fontFamilies';
+import { fontFamilies, nativeFontFamilies } from './fontFamilies';
 
-// TODO typography needs to be defined, after problems with fonts will be resolved by Ondra
-export const typography = {
-    title1: {
-        fontSize: fontSizes.title1,
-        lineHeight: lineHeights.small,
-        fontWeight: fontWeights.regular,
-        fontFamily: fontFamilies.base,
-    },
-    title2: {
-        fontSize: fontSizes.title2,
-        lineHeight: lineHeights.small,
-        fontWeight: fontWeights.regular,
-        fontFamily: fontFamilies.base,
-    },
-    title3: {
-        fontSize: fontSizes.title3,
-        lineHeight: lineHeights.small,
-        fontWeight: fontWeights.regular,
-        fontFamily: fontFamilies.base,
+export const fontWeights = {
+    medium: 400,
+    demi: 600,
+} as const;
+
+export type FontWeight = keyof typeof fontWeights;
+export type FontWeightValue = typeof fontWeights[FontWeight];
+
+// we need unit-less typography base because RN is unit-less, we can easily add units later
+// for web we need string instead of object because styled-components syntax
+export const typographyStylesBase = {
+    largeTitle: {
+        fontSize: 48,
+        lineHeight: 53,
+        fontWeight: fontWeights.medium,
+        letterSpacing: 0.4,
+    },
+    mediumTitle: {
+        fontSize: 34,
+        lineHeight: 37,
+        fontWeight: fontWeights.medium,
+        letterSpacing: -1.4,
+    },
+    smallTitle: {
+        fontSize: 22,
+        lineHeight: 26,
+        fontWeight: fontWeights.medium,
+        letterSpacing: -0.3,
+    },
+    highlight: {
+        fontSize: 16,
+        lineHeight: 24,
+        fontWeight: fontWeights.demi,
+        letterSpacing: -0.4,
+    },
+    body: {
+        fontSize: 16,
+        lineHeight: 24,
+        fontWeight: fontWeights.medium,
+        letterSpacing: -0.4,
+    },
+    callout: {
+        fontSize: 14,
+        lineHeight: 21,
+        fontWeight: fontWeights.demi,
+        letterSpacing: -0.3,
+    },
+    hint: {
+        fontSize: 14,
+        lineHeight: 21,
+        fontWeight: fontWeights.medium,
+        letterSpacing: -0.3,
+    },
+    label: {
+        fontSize: 12,
+        lineHeight: 18,
+        fontWeight: fontWeights.medium,
+        letterSpacing: -0.1,
     },
 } as const;
+
+export type TypographyStyle = keyof typeof typographyStylesBase;
+export type TypographyStyles = Record<TypographyStyle, string>;
+export type NativeTypographyStyleDefinition = {
+    fontSize: number;
+    lineHeight: number;
+    fontWeight: FontWeightValue;
+    letterSpacing: number;
+    fontFamily?: string;
+};
+export type NativeTypographyStyles = Record<TypographyStyle, NativeTypographyStyleDefinition>;
+
+export const prepareTypography = (): TypographyStyles => {
+    const typography: any = {};
+    // eslint-disable-next-line no-restricted-syntax
```

Comment:
why do we have this rule if it's already disabled 6 times in the codebase? If loops indeed put a dent on performance then why not simply use `.map()`?

Reply:
I think main concern of for loops is about immutability, not performance that could be probably even higher with for.

I am actually thinking about allowing it everywhere, because it could be quite helpful sometimes.

In the end I refactored it here to do not use loops.

Reply:
Never mind, I refactored function to not use hook so this variable is completely removed.